builder/proxmox/common/config.go

```diff
@@ -682,28 +683,6 @@ func (c *Config) Prepare(upper interface{}, raws ...interface{}) ([]string, []st
 		if disk.StoragePoolType != "" {
 			warnings = append(warnings, "storage_pool_type is deprecated and should be omitted, it will be removed in a later version of the proxmox plugin")
 		}
-		switch disk.Type {
-		case "ide":
-			ideCount++
-		case "sata":
-			sataCount++
-		case "scsi":
-			scsiCount++
-		case "virtio":
-			virtIOCount++
-		}
-	}
-	if ideCount > 2 {
-		errs = packersdk.MultiErrorAppend(errs, fmt.Errorf("maximum 2 IDE disks supported (ide2,3 reserved for ISOs)"))
-	}
-	if sataCount > 6 {
-		errs = packersdk.MultiErrorAppend(errs, fmt.Errorf("maximum 6 SATA disks supported"))
-	}
-	if scsiCount > 31 {
-		errs = packersdk.MultiErrorAppend(errs, fmt.Errorf("maximum 31 SCSI disks supported"))
-	}
-	if virtIOCount > 16 {
-		errs = packersdk.MultiErrorAppend(errs, fmt.Errorf("maximum 16 VirtIO disks supported"))
```

Comment:
Is there a reason to remove those checks? I suspect people may not hit the limit too easily, but it might be a good idea to error at this step so it doesn't error later during the build process (no cleanup needed and it happens before the build starts if it errors during `Prepare`

Reply:
Note: this is moved to `iso/config.go`, might be good enough, assuming the `clone` builder is not also using this? Otherwise we risk this not picking up on invalid configs for that other builder.